src/main/java/innohack/gem/service/extract/CSVExtractor.java

```diff
@@ -7,40 +7,48 @@
 import innohack.gem.entity.extractor.TimestampColumn;
 import innohack.gem.entity.feature.AbstractFeature;
 import innohack.gem.entity.feature.CsvFeature;
+import innohack.gem.entity.rule.ParamType;
+import innohack.gem.entity.rule.Parameter;
 import java.text.ParseException;
+import java.util.Arrays;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class CSVExtractor extends AbstractExtractor {
 
-  public CSVExtractor(ExtractConfig extractConfig) {
-    this.setExtractConfig(extractConfig);
+  public CSVExtractor() {
+    this(null);
+  }
+
+  public CSVExtractor(String headers) {
+    setParams(
+        Lists.newArrayList(new Parameter("Column Names", "", ParamType.STRING_LIST, headers)));
   }
 
   @Override
-  public ExtractedRecords extract(GEMFile f) throws Exception {
+  public ExtractedRecords extract(GEMFile f, ExtractConfig extractConfig) throws Exception {
     f.extract();
     for (AbstractFeature feature : f.getData()) {
       if (feature instanceof CsvFeature) {
-        List<String> extractColumns = getExtractConfig().getColumnNames();
-        List<TimestampColumn> extractTSColumns = getExtractConfig().getTimestampColumns();
+        String[] paramValues = getParams().get(0).getValue().split(",");
```

Comment:
Does this mean that the first parameter is always the columnNames? If not, we may need to match the param name to handle it?

Reply:
It's base on CSVExtractor. currently CSVExtractor only specify one parameter (STRING_LIST type) and is should be fix. When we change parameter in CSVExtractor's parameter, need to remember to change it in the extract method as well